src/DTO/EventDTO.php

```diff
@@ -1,144 +1,10 @@
-<?php
-
-declare(strict_types=1);
-
-namespace Abrouter\Client\DTO;
-
-class EventDTO
-{
-    /**
-     * @var string|null
-     */
-    private $temporaryUserId;
-
-    /**
-     * @var string|null
-     */
-    private $userId;
-
-    /**
-     * @var string
-     */
-    private $event;
-
-    /**
-     * @var string|null
-     */
-    private $referrer;
-
-    /**
-     * @var string|null
-     */
-    private $tag;
-
-    /**
-     * @var array|null
-     */
-    private $meta;
-
-    /**
-     * @var string|null
-     */
-    private $ip;
-
-    /**
-     * @var string|null
-     */
-    private $created_at;
-
-    /**
-     * EventDTO constructor.
-     *
-     * @param string|null $temporaryUserId
-     * @param string|null $userId
-     * @param string      $event
-     * @param string|null $tag
-     * @param string|null $referrer
-     * @param array|null  $meta
-     * @param string|null $ip
-     * @param string|null $created_at
-     */
-    public function __construct(
-        ?string $temporaryUserId = null,
-        ?string $userId = null,
-        string $event,
-        ?string $tag = null,
-        ?string $referrer = null,
-        ?array $meta = null,
-        ?string $ip = null,
-        ?string $created_at = null
-    ) {
-        $this->temporaryUserId = $temporaryUserId;
-        $this->userId = $userId;
-        $this->event = $event;
-        $this->tag = $tag;
-        $this->referrer = $referrer;
-        $this->meta = $meta;
-        $this->ip = $ip;
-        $this->created_at = $created_at;
-    }
-
-    /**
-     * @return string|null
-     */
-    public function getTemporaryUserId(): ?string
-    {
-        return $this->temporaryUserId;
-    }
-
-    /**
-     * @return string|null
-     */
-    public function getUserId(): ?string
-    {
-        return $this->userId;
-    }
-
-    /**
-     * @return string
-     */
-    public function getEvent(): string
-    {
-        return $this->event;
-    }
-
-    /**
-     * @return string|null
-     */
-    public function getReferrer(): ?string
-    {
-        return $this->referrer;
-    }
-
-    /**
-     * @return string|null
-     */
-    public function getTag(): ?string
-    {
-        return $this->tag;
-    }
-
-    /**
-     * @return array|null
-     */
-    public function getMeta(): ?array
-    {
-        return $this->meta;
-    }
-
-    /**
-     * @return string|null
-     */
-    public function getIp(): ?string
-    {
-        return $this->ip;
-    }
-
-    /**
-     * @return string|null
-     */
-    public function getCreatedAt(): ?string
-    {
-        return $this->created_at;
-    }
-}
+<?php
+
+declare(strict_types=1);
+
+namespace Abrouter\Client\DTO;
+
+interface EventDTO
+{
+    public function getBaseEventDTO (): BaseEventDTO;

```

Comment:
Everything is cool, but there is still one desire for me. Please, make it work as in the previous variant because we have a lot of code examples and we will rework it, but a bit later. 

I think we need to implement EventDTO builder for easy creation of this object because there are a lot of parameters and we can't even change the order without breaking our client's code. 

Then, please read carefully.

Can you please, rename EventDTO interface to EventDTOInterface and save this class as it WAS, but make it implement EventDTOInterface,

So, the old EventDTO will implement the new EventDTOInterface.

Because we need to save interface. No one knows who already implemented it in apps.

Probably new EventDTO will looks like:

class EventDTO implements EventDTOInterface {
public function __construct(?string $temporaryUserId = null, ?string $userId = null, ...) {
 $this->event = new BaseEventDTO(
$temporaryUserId,
$userId,
)
}

public function getBaseEvent(): BaseEvent{
return $this->event;
}

}